crates/store/re_chunk_store/src/query.rs

```diff
@@ -413,3 +560,201 @@ impl ChunkStore {
+            )
+    }
+
+    /// Returns the min and max times at which data was logged for an entity
+    /// on a specific timeline.
+    pub fn time_range_for_entity_on_timeline(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> Option<ResolvedTimeRange> {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let temporal_chunk_ids_per_timeline =
+            self.temporal_chunk_ids_per_entity.get(entity_path)?;
+        let chunk_id_sets = temporal_chunk_ids_per_timeline.get(timeline)?;
+
+        let start = chunk_id_sets.per_start_time.first_key_value()?.0;
+        let end = chunk_id_sets.per_end_time.last_key_value()?.0;
+
+        Some(ResolvedTimeRange::new(*start, *end))
+    }
+
+    /// Returns the number of events logged for an entity on a specific timeline.
+    pub fn num_events_on_timeline_for_all_components(
+        &self,
+        timeline: &Timeline,
+        entity_path: &EntityPath,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        let mut total_events = 0;
+
+        if let Some(static_chunks_per_component) = self.static_chunk_ids_per_entity.get(entity_path)
+        {
+            for chunk in static_chunks_per_component
+                .values()
+                .filter_map(|id| self.chunks_per_chunk_id.get(id))
+            {
+                total_events += chunk.num_events_cumulative();
+            }
+        }
+
+        if let Some(chunk_ids) = self
+            .temporal_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|temporal_chunks_events_per_timeline| {
+                temporal_chunks_events_per_timeline.get(timeline)
+            })
+        {
+            for chunk in chunk_ids
+                .per_start_time
+                .values()
+                .flat_map(|ids| ids.iter().filter_map(|id| self.chunks_per_chunk_id.get(id)))
+            {
+                total_events += chunk.num_events_cumulative();
+            }
+        }
+
+        total_events
+    }
+
+    /// Returns the number of times a static component was logged to an entity.
+    pub fn num_events_for_static_component(
+        &self,
+        entity_path: &EntityPath,
+        component_name: ComponentName,
+    ) -> usize {
+        re_tracing::profile_function!();
+
+        self.query_id.fetch_add(1, Ordering::Relaxed);
+
+        if let Some(static_chunk) = self
+            .static_chunk_ids_per_entity
+            .get(entity_path)
+            .and_then(|static_chunks_per_component| {
+                static_chunks_per_component.get(&component_name)
+            })
+            .and_then(|chunk_id| self.chunks_per_chunk_id.get(chunk_id))
+        {
+            static_chunk
+                .num_events_for_component(component_name)
+                .unwrap_or(0)
+        } else {
+            0
+        }
```

Comment:
Nah ok that one is pushing it too far imperativeness wise :D

```suggestion
        self.static_chunk_ids_per_entity
            .get(entity_path)
            .and_then(|static_chunks_per_component| {
                static_chunks_per_component.get(&component_name)
            })
            .and_then(|chunk_id| self.chunks_per_chunk_id.get(chunk_id))
            .and_then(|chunk| chunk.num_events_for_component(component_name))
            .unwrap_or(0)